### PR TITLE
gh-91217: deprecate spwd

### PR DESCRIPTION
**Pull request by brettcannon**
Automerge-Triggered-By: GH:brettcannon

### CODE REVIEW

File: Modules/spwdmodule.c [hunk 1]
```diff
@@ -256,5 +256,12 @@ static struct PyModuleDef spwdmodule = {
 PyMODINIT_FUNC
 PyInit_spwd(void)
 {
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                     "'nis' is deprecated and slated for removal in "
```
Comment (hugovk) [inline Modules/spwdmodule.c]:
```suggestion
                     "'spwd' is deprecated and slated for removal in "
```

File: Modules/spwdmodule.c [hunk 2]
```diff
@@ -256,5 +256,12 @@ static struct PyModuleDef spwdmodule = {
...
+                     "Python 3.13",
+                     7)) {
```
Comment (tiran) [inline Modules/spwdmodule.c:262]: Why do you use stack level 7 ?

Comment (brettcannon) [reply Modules/spwdmodule.c:262]: It seems to be what it takes for the traceback to show the import statement and not some internal detail of importlib.

File: Modules/spwdmodule.c (hunk 1 above)
Comment (brettcannon) [reply Modules/spwdmodule.c]: 😅

**Pull request merged**